### PR TITLE
Bunch o' hostile mobs fixes and refactoring

### PR DESCRIPTION
**Pull request by DeityLink**
- Renamed "metroid.dm" to "slime.dm" and renamed its "metroid" folder to "slime".
- Removed the obsolete hostile mob "IsInvalidTarget()" proc, the few times it was used have been replaced to use "CanAttack()" instead.

---
- Fixed gold slime monsters attacking adamantine golems.
- Improved a bunch of code related to hostile mob target acquisition.
- Rabid slimes now remember their friends and don't attack them. This should ease the task of good xenobiologists, and even add more use to rabid slimes.
- Simple Animals can now deal damage other than brute.
- Rabid slimes for instance now deal clone damage instead of brute. They also now move on top of their target.
- Simple Animal Aliens now take double damage from burn sources.
- Simple Animal Aliens now use the proper deaths sprites, instead of the deprecated sideway ones.
- Simple Animal Aliens can finally be meat-spiked.
- Simple Animal Aliens no longer attack Carbon Aliens.

### CODE REVIEW

**Comment by ConkerDev** [comment]
:+1: :love_hotel:

File: code/modules/mob/living/carbon/alien/humanoid/humanoid.dm [hunk 1]
```diff
@@ -178,7 +178,11 @@
 			playsound(loc, M.attack_sound, 50, 1, 1)
 		visible_message("<span class='warning'><B>[M]</B> [M.attacktext] \the [src] !</span>")
 		var/damage = rand(M.melee_damage_lower, M.melee_damage_upper)
-		adjustBruteLoss(damage)
+		var/dam_zone = pick("chest", "l_hand", "r_hand", "l_leg", "r_leg")
```
Comment (clusterfack) [inline code/modules/mob/living/carbon/alien/humanoid/humanoid.dm]: If you were going to copypaste it you should have just made a proc for it

**Comment by clusterfack** [comment]
You fell for the old 'rename files as you change something with the file' trick. Resulting in every single line being counted as a line changed.

File: code/modules/mob/living/simple_animal/hostile/slime.dm [hunk 2]
```diff
@@ -59,95 +58,117 @@
...
+		forceMove(get_turf(target))
+
+/mob/living/simple_animal/hostile/slime/AttackingTarget()
+	forceMove(get_turf(target))
+	..()
+
 /mob/living/simple_animal/hostile/slime/proc/calm()
 	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/mob/living/simple_animal/hostile/slime/proc/calm() called tick#: [world.time]")
+	var/calmed_type = /mob/living/carbon/slime
 	switch(colour)
```
Comment (clusterfack) [inline code/modules/mob/living/simple_animal/hostile/slime.dm]: It's funny, if the colours perfectly matched the typepath you could just use

text2path("/mob/living/carbon/slime/" + colour)

**Comment by DeityLink** [comment]
>You fell for the old 'rename files as you change something with the file' trick. Resulting in every single line being counted as a line changed.

@clusterfack which is why I renamed the files/folder in a separate commit ( be270d6 )

File: code/modules/mob/living/simple_animal/hostile/slime.dm (hunk 2 above)
Comment (DeityLink) [inline code/modules/mob/living/simple_animal/hostile/slime.dm]: oh that's a thing? that's nice

**Comment by IratePirate** [comment]
>Renamed "metroid.dm" to "slime.dm" and renamed its "metroid" folder to "slime".

Why don't you do the right thing and bring back the Metroid sprites instead to make up for TG's weakness.

**Comment by DeityLink** [comment]
@IratePirate completely separating slimes from metroids is the first step if we want metroids to come back at all.

**Comment by N3X15** [comment]
Is this good to go @clusterfack @DeityLink

**Comment by clusterfack** [comment]
Besides my final comment this seems fine

**Comment by DeityLink** [comment]
@N3X15 Yeah it's good to go now

**Pull request merged**